src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -30,53 +37,108 @@
 import org.slf4j.LoggerFactory;
 
 public class ClipBoardManager {
+
     public static final DataFormat XML = new DataFormat("application/xml");
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ClipBoardManager.class);
 
-    private final Clipboard clipboard;
-    private final ImportFormatReader importFormatReader;
+    private static Clipboard clipboard;
+    private static java.awt.datatransfer.Clipboard primary;
+    private static ImportFormatReader importFormatReader;
 
-    public ClipBoardManager() {
-        this(Clipboard.getSystemClipboard(), Globals.IMPORT_FORMAT_READER);
+    public ClipBoardManager(Clipboard clipboard, java.awt.datatransfer.Clipboard primary, ImportFormatReader importFormatReader) {
+        ClipBoardManager.clipboard = clipboard;
+        ClipBoardManager.primary = primary;
+        ClipBoardManager.importFormatReader = importFormatReader;
     }
 
-    public ClipBoardManager(Clipboard clipboard, ImportFormatReader importFormatReader) {
-        this.clipboard = clipboard;
-        this.importFormatReader = importFormatReader;
+    public ClipBoardManager() {
```

Comment:
Why were the two constructors swapped? Could you keep them in the same order as they were?

Reply:
D'oh! Sorry, I didn't realize that I made that change. I'll revert it.